src/qt/nametablemodel.cpp

```diff
@@ -0,0 +1,537 @@
+        }
+    }
+    return QVariant();
+}
+
+QVariant
+NameTableModel::headerData(int section, Qt::Orientation orientation, int role) const
+{
+    if (orientation == Qt::Horizontal)
+    {
+        if (role == Qt::DisplayRole)
+        {
+            return columns[section];
+        }
+        else if (role == Qt::TextAlignmentRole)
+        {
+            return column_alignments[section];
+        }
+        else if (role == Qt::ToolTipRole)
+        {
+            switch(section)
+            {
+            case Name:
+                return tr("Name registered using Namecoin.");
+            case Value:
+                return tr("Data associated with the name.");
+            case ExpiresIn:
+                return tr("Number of blocks, after which the name will expire."
+                          " Update name to renew it.\nEmpty cell means pending"
+                          " (awaiting automatic name_firstupdate or awaiting "
+                          "network confirmation).");
+            }
+        }
+    }
+    return QVariant();
+}
+
+Qt::ItemFlags
+NameTableModel::flags(const QModelIndex &index) const
+{
+    if (!index.isValid())
+        return 0;
+    //NameTableEntry *rec = static_cast<NameTableEntry*>(index.internalPointer());
+
+    return Qt::ItemIsSelectable | Qt::ItemIsEnabled;
+}
+
+QModelIndex
+NameTableModel::index(int row, int column, const QModelIndex &parent /* = QModelIndex()*/) const
+{
+    Q_UNUSED(parent);
+    NameTableEntry *data = priv->index(row);
+    if (data)
+    {
+        return createIndex(row, column, priv->index(row));
+    }
+    else
+    {
+        return QModelIndex();
+    }
+}
+
+// queue notifications to show a non freezing progress dialog e.g. for rescan
+struct TransactionNotification
+{
+public:
+    TransactionNotification() {}
+    TransactionNotification(uint256 hash, ChangeType status, bool showTransaction):
+        hash(hash), status(status), showTransaction(showTransaction) {}
+
+    void invoke(NameTableModel *ntm)
+    {
+        QString strHash = QString::fromStdString(hash.GetHex());
+        QMetaObject::invokeMethod(ntm, "updateTransaction", Qt::QueuedConnection,
+                                  Q_ARG(QString, strHash),
+                                  Q_ARG(int, status));
+    }
+private:
+    uint256 hash;
+    ChangeType status;
+    bool showTransaction;
+};
+
+static bool fQueueNotifications = false;
+static std::vector< TransactionNotification > vQueueNotifications;
+
+static void NotifyTransactionChanged(NameTableModel *ntm, CWallet *wallet, const uint256 &hash, ChangeType status)
+{
+    // Find transaction in wallet
+    std::map<uint256, CWalletTx>::iterator mi = wallet->mapWallet.find(hash);
+    // Determine whether to show transaction or not (determine this here so that no relocking is needed in GUI thread)
+    bool inWallet = mi != wallet->mapWallet.end();
+    // bool showTransaction = (inWallet && TransactionRecord::showTransaction(mi->second));
```

Comment:
Again, please remove commented-out code or clarify in a comment why it is still there.